### PR TITLE
Applications

### PR DESCRIPTION
**Pull request by takeller**
- Finished all application stories
- Refactor opportunities with approving an application
     - Should maybe pull some logic into the model 
     - Application show page currently uses the .any? enumerable 
- All tests passing

### CODE REVIEW

File: app/controllers/applications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    if params[:favorite_pet_ids].nil?
+      redirect_to "/applications/new"
+      flash[:notice] = "All Fields Required!"
+      return
+    end
+
+    if app.save
+      params[:favorite_pet_ids].each do |favorite_pet_id|
+        favorite_pet_id = favorite_pet_id.to_i
+        PetApplication.create!({pet_id: favorite_pet_id, application_id: app.id})
```
Comment (takeller) [inline app/controllers/applications_controller.rb:33]: Should probably not have the ! on create

File: app/controllers/pets_controller.rb [hunk 2]
```diff
@@ -25,21 +25,38 @@ def edit
 
   def update
     pet = Pet.find(params[:id])
+    if pet.adoption_status == "Pending"
+      flash[:notice] = 'Some message'
```
Comment (takeller) [inline app/controllers/pets_controller.rb:29]: Need to update with correct message

File: app/controllers/shelters_controller.rb [hunk 3]
```diff
@@ -53,13 +53,22 @@ def update
 
   def destroy
     shelter = Shelter.find(params[:id])
-    shelter.reviews.destroy
-    shelter.pets.destroy
+    approved_pets = shelter.pets.find_all do |pet|
+      pet.pet_applications.any? {|pet_app| pet_app.approved}
```
Comment (takeller) [inline app/controllers/shelters_controller.rb:57]: Maybe a better way to do this?

File: app/views/applications/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+<p> State: <%= @application.state %> </p>
+<p> Zip: <%= @application.zip %> </p>
+<p> Phone: <%= @application.phone %> </p>
+<p> Description: <%= @application.description %> </p>
+
+<h4>Pets Applied For</h4>
+<%= form_tag("/applications/#{@application.id}/multi_approve", method: :patch) do %>
+  <% @application.pet_applications.each do |pet_app| %>
+    <section id="pet-<%=pet_app.pet.id%>">
+      <% if pet_app.pet.pet_applications.any? { |pet_app| pet_app.approved} %>
```
Comment (takeller) [inline app/views/applications/show.html.erb:15]: I don't like having this .any? here. How can we pull it out of the view?

File: app/controllers/applications_controller.rb [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  end
+
+  def new
+    favorite = Favorite.new(session[:favorite])
+    @pets = favorite.id_to_object
+  end
+
+  def show
+    @application = Application.find(params[:id])
+    # binding.pry
```
Comment (Ashkanthegreat) [inline app/controllers/applications_controller.rb:17]: Lets remove this in refactoring.

File: app/controllers/applications_controller.rb [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    # @pet_apps = @application.pet_applications.all
```
Comment (Ashkanthegreat) [inline app/controllers/applications_controller.rb:18]: Also remove this in refactoring.

File: app/views/pets/index.html.erb [hunk 7]
```diff
@@ -1,15 +1,19 @@
 <h1>All Pets</h1>
 
 <% @pets.each do |pet| %>
+<section id="pet-<%=pet.id%>">
   <div class="petContainer">
-    <%= image_tag(pet.image, size: "300x250") %>
+    <%= image_tag(pet.image, size: "300x300") %>
```
Comment (Ashkanthegreat) [inline app/views/pets/index.html.erb:6]: Let's change this to auto size so the pics don't look stretched.

**Pull request merged**